x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/transport/actions/put/UpdateWatcherSettingsAction.java

```diff
@@ -34,6 +34,16 @@ public class UpdateWatcherSettingsAction extends ActionType<AcknowledgedResponse
         IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS
     );
 
+    public static final Set<String> ALLOWED_SETTINGS_PREFIXES = Set.of(
+        IndexMetadata.INDEX_ROUTING_EXCLUDE_GROUP_PREFIX,
+        IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_PREFIX,
+        IndexMetadata.INDEX_ROUTING_REQUIRE_GROUP_PREFIX
+    );
+
+    public static final Set<String> EXPLICITLY_DENIED_SETTINGS = Set.of(
```

Comment:
I assume you're denying this one because we always want the .watches shards in the hot tier? That makes me wonder what other possible values there are for these afix settings, and if we ought to just explicitly allow the `index.routing.allocation.exclude.role`, `index.routing.allocation.include.role`, and `index.routing.allocation.require.role` ones and disallow anything else to avoid causing some unforeseen problems. You've probably looked at all these settings more than I have -- what do you think?

Reply:
So I based this restriction on what the default setting for the `.watcher` index are. I assumed we put it in the hot tier for good reason so locked off the ability to change that.

Locking down the `index.routing.allocation.*.*` settings further is difficult. `index.routing.allocation.include.role` isn't actually a setting as such but a convention for a wildcard setting. Anything after `index.routing.allocation.include.` is assumed to be an arbitrary node attribute hence my leaving it as allowing any setting with that prefix as we have no way of knowing what attributes the end user might have set on their nodes and how they would like to use those for scheduling watches. 

For instance `index.routing.allocation.include.sandwich_filling=turkey` is a valid setting for this if the user chose to use a bread based node attributes scheme to group their cluster members :-)